assets/src/features/room-page/components/PinComponents.tsx

```diff
@@ -10,25 +10,48 @@ type PinUserButtonProps = {
 
 export const PinTileLayer: FC<PinUserButtonProps> = ({ pinned, onClick }: PinUserButtonProps) => {
   const pinText = pinned ? "Unpin" : "Pin";
+  const [showLayer, setShowLayer] = useState<boolean>(false);
+  const timeRef = useRef<NodeJS.Timeout | null>(null);
+
+  const restartTimer = useCallback(() => {
+    const fiveSeconds = 5_000;
+
+    if (timeRef.current) {
+      clearTimeout(timeRef.current);
+    }
+
+    setShowLayer(true);
+    timeRef.current = setTimeout(() => {
+      setShowLayer(false);
+    }, fiveSeconds);
+  }, []);
+
+  useEffect(() => {
+    return () => {
+      if (timeRef.current) clearTimeout(timeRef.current);
+    };
+  }, []);
 
   return (
-    <div className="group absolute h-full w-full">
-      <Button
-        className={clsx(
-          "absolute top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2",
-          "hidden group-hover:flex",
-          "flex-row content-center",
-          "opacity-50 hover:opacity-75",
-          "gap-2 py-4 px-8",
-          "rounded-full bg-brand-grey-100"
-        )}
-        onClick={onClick}
-      >
-        <Pin className="text-brand-white" />
-        <span className={"font-lg font-aktivGrotesk font-semibold leading-6 tracking-wide text-brand-white"}>
-          {pinText}
-        </span>
-      </Button>
+    <div className={"group absolute h-full w-full"} onMouseMove={restartTimer}>
```

Comment:
I thought this would have little or no effect since I had been able to use the site without any problems when implementing this feature that way. Unfortunately, I have just tested this feature more rigorously and the `performance` tab shows this feature consumes large amounts of CPU and can cause small frame drops when rendering. I'll investigate this further and will compare to videoconference web apps with a similar feature.

Reply:
<img width="963" alt="screenshoot" src="https://user-images.githubusercontent.com/19818474/223437466-f8716c81-98fa-4c00-b1f7-04be3643afb6.png">

The blue box represents hovering over the video tile, while the green box represents hovering over the copy link component. That mouse action should not have a performance impact.